openassessment/xblock/static/js/src/lms/oa_grade.js

```diff
@@ -16,15 +16,21 @@ OpenAssessment.GradeView.prototype = {
     /**
      * Load the grade view.
      */
-    load: function() {
+    load: function(usageID) {
         var view = this;
         var baseView = this.baseView;
+        var stepID = "#openassessment__grade";
         this.server.render('grade').done(
             function(html) {
                 // Load the HTML and install event handlers
-                $('#openassessment__grade', view.element).replaceWith(html);
-                view.server.renderLatex($('#openassessment__grade', view.element));
+                $(stepID, view.element).replaceWith(html);
+                view.server.renderLatex($(stepID, view.element));
                 view.installHandlers();
+                if (typeof usageID !== 'undefined' &&
```

Comment:
Yes - I don't pass in usageID when the steps are all initially loading, when you first go to the page.

This function in all the steps is called on first load of the page and any time the user makes a change. Because all the logic for the state of each step is performed on the back end, I need some way on the front end to send focus to the appropriate step's button when the user makes a change.

Initially, without sending a usageID, I would just check for whether the step is "current" (whether it should be opened by default or not) and if it is, send the focus to that step. If I had multiple ORA's on a single page, there will be multiple "current" steps, so the ID needs to be passed in so we know which ID to send focus to.

But if I pass in the usageID upon first loading the page, the focus jumps around from current step to current step, when I just want it at the top of the page on first load. So I don't pass in the usageID when first loading the page, but I do pass it in every time afterwards.